test/plugin/test_out_google_cloud.rb

```diff
@@ -463,6 +482,35 @@ def stub_metadata_request(metadata_path, response_body)
                 :headers => {'Content-Length' => response_body.length})
   end
 
+  def setup_auth_stubs
+    # Used by 'googleauth' to test whether we're running on GCE.
+    # It only cares about the request succeeding with Metdata-Flavor: Google.
+    stub_request(:get, 'http://169.254.169.254').
```

Comment:
It is a defined well-known address for the metadata server: https://cloud.google.com/compute/docs/metadata

It's hardcoded that way in https://github.com/google/google-auth-library-ruby/blob/b5ff47fecb009f45262ee7ab87d294e4ccb1e7c1/lib/googleauth/compute_engine.rb#L56 whereas other libraries hardcode http://metadata/, but stub_request is only looking for an exact match.

Perhaps I should add a method so we register all these stubs on both places (actually, per the docs we should include http://metadata.google.internal/ as well), but let's do that in another CL.